src/model/openthaigpt_pretraining_model/lightning/utils.py

```diff
@@ -120,17 +121,36 @@ def __init__(
             )
         else:
             raise NotImplementedError("only support Llama, llama_hf or GPTJ")
-        train_dataset = get_dataset(dataset_name, split=SPLIT_TRAIN, shuffle=True)
```

Comment:
You should not delete this.

Reply:
what about streaming dataset?

Reply:
We should handle `streaming dataset` in `get_dataset`.